### ISSUE TITLE
Add devcontainer for all services

### ISSUE THREAD

**Issue by mochiruntime** [clicepfl/website#6]
Configure devcontainers for the monorepository structure: when developers open the project with VSCode, it should start all the services automatically

- Follow the https://containers.dev/ specification
- Because VSCode is not able to open and connect to multiple devcontainers simultaneously, we cannot use workspaces.
- Opening the entire repository folder may be awkward for some tools: for instance the VSCode Prettier plugin expects .prettierrc to be at the root, and does not play well with an environment with subprojects.
- The devcontainer will work as follows: opening any of the subproject should launch a top-level docker-compose. However, each devcontainer definition is local to its subproject. This allows different parts of the project to have completely different container and editor settings.

Todo:
- [ ] It is possible to connect to multiple containers through the remote extensions panel, right-click on a container and connect, this opens in a new VSCode window. Check if closing any of these windows shuts down the entire stack while previous windows are still open.

**Comment by DeepGreen1** [comment]
LGTM

**Comment by NoeTerrier** [comment]
When opening folder in a container with command :

`/snap/code/115/usr/share/code/code --ms-enable-electron-run-as-node /home/noe/.vscode/extensions/ms-vscode-remote.remote-containers-0.266.1/dist/spec-node/devContainersSpecCLI.js read-configuration --workspace-folder /home/noe/CLIC/clic-website-v2/strapi/ --log-level debug --log-format json --config /home/noe/CLIC/clic-website-v2/strapi/.devcontainer/devcontainer.json --include-merged-configuration --mount-workspace-git-root true`

I got error:

```
[7845:1219/191319.851175:ERROR:viz_main_impl.cc(186)] Exiting GPU process due to errors during initialization
[7852:1219/191319.890568:ERROR:angle_platform_impl.cc(43)] Display.cpp:977 (initialize): ANGLE Display::initialize error 12289: Could not dlopen libGL.so.1: /snap/core/current/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /lib/x86_64-linux-gnu/libGLdispatch.so.0)
[7852:1219/191319.890681:ERROR:gl_surface_egl.cc(852)] EGL Driver message (Critical) eglInitialize: Could not dlopen libGL.so.1: /snap/core/current/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /lib/x86_64-linux-gnu/libGLdispatch.so.0)
[7852:1219/191319.890750:ERROR:gl_surface_egl.cc(1489)] eglInitialize OpenGL failed with error EGL_NOT_INITIALIZED, trying next display type
[7852:1219/191319.891098:ERROR:angle_platform_impl.cc(43)] Display.cpp:977 (initialize): ANGLE Display::initialize error 12289: Could not dlopen libGL.so.1: /snap/core/current/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /lib/x86_64-linux-gnu/libGLdispatch.so.0)
[7852:1219/191319.891169:ERROR:gl_surface_egl.cc(852)] EGL Driver message (Critical) eglInitialize: Could not dlopen libGL.so.1: /snap/core/current/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /lib/x86_64-linux-gnu/libGLdispatch.so.0)
[7852:1219/191319.891266:ERROR:gl_surface_egl.cc(1489)] eglInitialize OpenGLES failed with error EGL_NOT_INITIALIZED
[7852:1219/191319.891335:ERROR:gl_ozone_egl.cc(21)] GLSurfaceEGL::InitializeOneOff failed.
[7852:1219/191319.892237:ERROR:viz_main_impl.cc(186)] Exiting GPU process due to errors during initialization
Warning: 'workspace-folder' is not in the list of known options, but still passed to Electron/Chromium.
Warning: 'log-level' is not in the list of known options, but still passed to Electron/Chromium.
Warning: 'log-format' is not in the list of known options, but still passed to Electron/Chromium.
Warning: 'config' is not in the list of known options, but still passed to Electron/Chromium.
Warning: 'include-merged-configuration' is not in the list of known options, but still passed to Electron/Chromium.
Warning: 'mount-workspace-git-root' is not in the list of known options, but still passed to Electron/Chromium.
[7861:1219/191319.911184:ERROR:angle_platform_impl.cc(43)] Display.cpp:977 (initialize): ANGLE Display::initialize error 12289: Could not dlopen libGL.so.1: /snap/core/current/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /lib/x86_64-linux-gnu/libGLdispatch.so.0)
[7861:1219/191319.911297:ERROR:gl_surface_egl.cc(852)] EGL Driver message (Critical) eglInitialize: Could not dlopen libGL.so.1: /snap/core/current/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /lib/x86_64-linux-gnu/libGLdispatch.so.0)
[7861:1219/191319.911370:ERROR:gl_surface_egl.cc(1489)] eglInitialize OpenGL failed with error EGL_NOT_INITIALIZED, trying next display type
[7861:1219/191319.911706:ERROR:angle_platform_impl.cc(43)] Display.cpp:977 (initialize): ANGLE Display::initialize error 12289: Could not dlopen libGL.so.1: /snap/core/current/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /lib/x86_64-linux-gnu/libGLdispatch.so.0)
[7861:1219/191319.911787:ERROR:gl_surface_egl.cc(852)] EGL Driver message (Critical) eglInitialize: Could not dlopen libGL.so.1: /snap/core/current/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /lib/x86_64-linux-gnu/libGLdispatch.so.0)
[7861:1219/191319.911828:ERROR:gl_surface_egl.cc(1489)] eglInitialize OpenGLES failed with error EGL_NOT_INITIALIZED
[7861:1219/191319.911864:ERROR:gl_ozone_egl.cc(21)] GLSurfaceEGL::InitializeOneOff failed.
[7861:1219/191319.913229:ERROR:viz_main_impl.cc(186)] Exiting GPU process due to errors during initialization

```

**Comment by mochiruntime** [comment]
I can confirm that this bug affects multiple Linux environments (ubuntu, manjaro). It works properly on MacOS, and Windows should be fine as well (to be confirmed).

However I've tried using other devcontainers and they work just fine. The actual error seems unrelated to the GPU error messages.

@NoeTerrier can you open the devcontainer in this example? https://github.com/microsoft/vscode-remote-try-node#vs-code-dev-containers 

So the problem probably comes from our config. I don't understand why it only impacts Linux though :thinking:

**Comment by mochiruntime** [comment]
:tada: @NoeTerrier @Robb-Fr I have found a solution! You probably have an outdated version of the `docker-compose` utility.

Check if you have a binary at `/usr/local/bin/docker-compose`. If that's the case, you probably installed it using https://docs.docker.com/compose/install/other/. You should be able to remove it completely, since Docker now ships with `docker compose` included, so VSCode was probably first testing for `docker-compose` in path, and then falling back to the built-in compose. If you don't have the built-in plugin, have a look at https://docs.docker.com/compose/install/linux/ or reinstall the latest standalone at https://docs.docker.com/compose/install/other/.

**Comment by Lumycea** [comment]
On Manjaro 22.0.0:

It worked, but I had to open the container twice with VSCode (first time failed after build). I had neither `docker-compose` nor `docker compose` installed, and I had to install `docker-compose` through `pamac`. Now I have both, version 2.14.1.

But the container is disconnecting after a few seconds

**Comment by mochiruntime** [comment]
> It worked, but I had to open the container twice with VSCode (first time failed after build).

That's weird indeed. I have tried to open it anew after running `docker system prune --all` and it worked on the first try. Let me know if you find the issue.

> But the container is disconnecting after a few seconds

That is normal, you probably don't have a local `.env` file. This will be fixed by #8

**Comment by mochiruntime** [comment]
See #23 for updated devcontainer architecture
